### PR TITLE
Hub clients page no longer crashes when client intake is archived

### PR DESCRIPTION
**Pull request by tgrathwell**
We do need to hide a lot of controls that can edit the intake,
because in theory an archived intake is readonly

Co-authored-by: Asheesh Laroia <alaroia@codeforamerica.org>

### CODE REVIEW

File: app/models/client.rb [hunk 1]
```diff
@@ -227,6 +227,7 @@ def increment_failed_attempts
   end
 
   def requires_spouse_info?
+    return false unless intake # TODO presenter
```
Comment (alaroia-cfa) [inline app/models/client.rb:230]: TODO

File: app/models/client.rb [hunk 2]
```diff
@@ -288,7 +291,8 @@ def online_ctc?
   end
 
   def hub_status_updatable
-    !online_ctc?
+    # TODO: Probably move to the presenter
```
Comment (alaroia-cfa) [inline app/models/client.rb:294]: TODO

File: app/views/hub/messages/index.html.erb [hunk 3]
```diff
@@ -62,7 +62,7 @@
 
       <% if @client&.intake.email_address.present? %>
         <div class="single-input-form spacing-above-25">
-          <%= form_with model: @outgoing_email, url: hub_client_outgoing_emails_path(client: @client), local: true, builder: VitaMinFormBuilder, method: "post", html: {class: 'email-form'} do |f| %>
```
Comment (alaroia-cfa) [inline app/views/hub/messages/index.html.erb:65]: Can/should this be `@client.outgoing_emails.build`?

File: spec/factories/archived/intake_2021.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
```
Comment (alaroia-cfa) [inline spec/factories/archived/intake_2021.rb:1]: Probably should be renamed to be `spec/factories/archived/ctc_intake_2021.rb`

File: spec/features/hub/show_client_spec.rb [hunk 5]
```diff
@@ -48,6 +48,18 @@
         expect(page).not_to have_css(".tax-return-inline-form")
       end
     end
+
+    context "for a client with an archived 2021 intake" do
+      before do
+        client.intake.destroy
+        @intake = create(:archived_2021_intake, client: client)
```
Comment (alaroia-cfa) [inline spec/features/hub/show_client_spec.rb:55]: Would this be tidier as a `let(:intake) { nil }` then `let(:archived_2021_intake) { create(:archived_2021_intake, client: client}`?

**Review by alaroia-cfa — COMMENTED** [review]
Some comments; I know it's a draft

File: app/models/client.rb (hunk 2 above)
Comment (alaroia-cfa) [reply app/models/client.rb]: I wasn't able to get this done, and I can live with that.

File: spec/features/hub/show_client_spec.rb (hunk 5 above)
Comment (alaroia-cfa) [reply spec/features/hub/show_client_spec.rb:55]: Sure, let's try it

File: app/views/hub/messages/index.html.erb [hunk 6]
```diff
@@ -44,7 +44,7 @@
 
       <% if @client&.intake.sms_phone_number.present? %>
         <div class="single-input-form no-form-group-padding">
-          <%= form_with model: @client.outgoing_text_message, url: hub_client_outgoing_text_messages_path(client: @client), local: true, builder: VitaMinFormBuilder, method: "post", html: {class: 'text-message-form'} do |f| %>
+          <%= form_with model: @client.outgoing_text_messages.build, url: hub_client_outgoing_text_messages_path(client: @client), local: true, builder: VitaMinFormBuilder, method: "post", html: {class: 'text-message-form'} do |f| %>
```
Comment (tgrathwell) [inline app/views/hub/messages/index.html.erb:47]: SMART

**Pull request merged**